build_modules/lib/src/module_library.dart

```diff
@@ -127,6 +129,23 @@ class ModuleLibrary {
         hasMain: _hasMainMethod(parsed));
   }
 
+  static void _checkValidConditionalImport(
+      Uri? parsedUri, AssetId id, UriBasedDirective node) {
+    if (parsedUri == null) {
+      throw ArgumentError(
+          'Unsupported conditional import with non-constant uri found in $id:'
+          '\n\n${node.toSource()}');
+    } else if (parsedUri.scheme == 'dart') {
+      throw ArgumentError(
+          'Unsupported conditional import of `$parsedUri` found in $id:\n\n'
+          '${node.toSource()}\n\nThis environment does not support direct '
+          'conditional imports of `dart:` libraries. Instead you must create '
+          'a separate library which imports (or exports) the `dart:` library '
```

Comment:
consider `which unconditionally imports (or exports) the \`dart:\` library` to make it crystal clear.